google/cloud/pubsub/internal/subscription_session_test.cc

```diff
@@ -305,6 +306,130 @@ TEST(SubscriptionSessionTest, ScheduleCallbacksExactlyOnce) {
   for (auto& t : tasks) t.join();
 }
 
+/// @test Verify callbacks are scheduled in the background threads with Open
+/// Telemetry enabled.
+TEST(SubscriptionSessionTest, ScheduleCallbacksWithOtelEnabled) {
```

Comment:
What are you testing? That the library doesn't crash?

There is an observable behavior of having the option set to true. The message callback is traced. We should be able to capture spans given the right conditions.

This test is >100 lines of code. It is way too complicated for this purpose. Try to write the simplest test that exercises the code under test.

Reply:
Done